src/binary/decimal.rs

```diff
@@ -108,24 +109,38 @@ where
 
     fn encode_decimal_value(&mut self, decimal: &Decimal) -> IonResult<usize> {
         let mut bytes_written: usize = 0;
-        // First encode the decimal. We need to know the encoded length before
-        // we can compute and write out the type descriptor.
+        // First encode the decimal value to a stack-allocated buffer.
```

Comment:
Just curious—is it possible (and useful) to calculate the size prior to encoding it?